modules/katamari/src/main/ts/ephox/katamari/api/Type.ts

```diff
@@ -1,27 +1,26 @@
 const getPrototypeOf = Object.getPrototypeOf;
 
-const hasProto = (v: Object, predicate: (v: Object) => boolean, name: string): boolean => {
-  if (predicate(v)) {
+interface Constructor<T extends Object> {
+  readonly prototype: T;
+  readonly name: string;
+}
+
+const hasProto = <T extends Object>(v: Object, constructor: Constructor<T>, predicate: (v: Object, prototype: T) => boolean): boolean => {
+  if (predicate(v, constructor.prototype)) {
     return true;
   } else {
     // String-based fallback time
-    const constructor = v.constructor;
-    if (isNonNullable(constructor)) {
-      return constructor.name === name;
-    } else {
-      // IE doesn't support the constructor API
-      return getPrototypeOf(v).toString() === `[object ${name}]`;
-    }
+    return v.constructor?.name === constructor.name;
   }
 };
 
 const typeOf = (x: any): string => {
   const t = typeof x;
   if (x === null) {
     return 'null';
-  } else if (t === 'object' && hasProto(x, (v) => Array.prototype.isPrototypeOf(v), 'Array')) {
+  } else if (t === 'object' && hasProto(x, Array, (o, proto) => proto.isPrototypeOf(o))) {
```

Comment:
I've been meaning to change this implementation to just `Array.isArray()` for a while. It works even for cross-window arrays:
https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/Array/isArray#instanceof_vs_isarray

Reply:
:+1: I'll change it as part of this then